src/AssignSubstitutes/InputOutput/XMLParser.java

```diff
@@ -43,42 +49,220 @@
 
     private String filepath;
 
-    public XMLParser() {
-        Document doc = null;
-
-        //TODO: add defaults
-        try
-        { doc = getDocument("config"); }
-        catch (TransformerException | SAXException | ParserConfigurationException | IOException e)
-        { e.printStackTrace(); }
-
-        NodeList config = doc.getElementsByTagName("config");
-        Element settings = (Element)config.item(0);
+    public XMLParser() throws TransformerException, SAXException, ParserConfigurationException, IOException
+    {
+        Document doc;
+        Element settings;
 
         this.filepath = "config";
 
-        //TODO: make not shit the bed whenever something doesn't exist
-
-        this.maxWeeklyTally = Integer.valueOf(settings.getElementsByTagName("weeklyMax").item(0).getTextContent());
-        this.maxMonthlyTally = Integer.valueOf(settings.getElementsByTagName("monthlyMax").item(0).getTextContent());
-        this.tempWeeklyMax = Integer.valueOf(settings.getElementsByTagName("tempWeeklyMax").item(0).getTextContent());
-        this.tempMonthlyMax = Integer.valueOf(settings.getElementsByTagName("tempMonthlyMax").item(0).getTextContent());
-
-        this.startDate = LocalDate.parse(settings.getElementsByTagName("startDate").item(0).getTextContent());
-        this.weeksToReminder = Integer.valueOf(settings.getElementsByTagName("weeksUntilEnd").item(0).getTextContent());
+        try {
+            doc = getDocument(filepath);
+
+            NodeList config = doc.getElementsByTagName("config");
```

Comment:
getElementsByTagName throws a nullPointerException so if it's in the try catch then it should catch that too.